prac_07/language_file_reader.py

```diff
@@ -13,76 +13,51 @@
 def main():
     """Read file of programming language details, save as objects, display."""
     languages = []
-    # Open the file for reading
-    in_file = open('languages.csv', 'r')
-    # File format is like: Language,Typing,Reflection,Year
-    # 'Consume' the first line (header) - we don't need its contents
-    in_file.readline()
-    # All other lines are language data
-    for line in in_file:
-        # print(repr(line))  # debugging
-        # Strip newline from end and split it into parts (CSV)
-        parts = line.strip().split(',')
-        # print(parts)  # debugging
-        # Reflection is stored as a string (Yes/No) and we want a Boolean
-        reflection = parts[2] == "Yes"
-        # Construct a ProgrammingLanguage object using the elements
-        # year should be an int
-        language = ProgrammingLanguage(parts[0], parts[1], reflection, int(parts[3]))
-        # Add the language we've just constructed to the list
-        languages.append(language)
-    # Close the file as soon as we've finished reading it
-    in_file.close()
-
-    # Loop through and display all languages (using their str method)
+    with open('languages.csv', 'r') as in_file:
+        in_file.readline()  # Skip header
+        for line in in_file:
+            parts = line.strip().split(',')
+            reflection = parts[2] == "Yes"
+            pointer_arithmetic = parts[4] == "Yes"
+            language = ProgrammingLanguage(parts[0], parts[1], reflection, int(parts[3]), pointer_arithmetic)
+            languages.append(language)
+
+    print("Languages that support pointer arithmetic:")
     for language in languages:
-        print(language)
+        if language.is_arithmetic():
+            print(language)
 
 
 main()
 
 
 def using_csv():
     """Language file reader version using the csv module."""
-    # First, open the file for reading - note: specify newline
-    # to avoid quoted \n in strings being considered a new record
-    in_file = open('languages.csv', 'r', newline='')
-    in_file.readline()
-    reader = csv.reader(in_file)  # use default dialect, Excel
-    for row in reader:
-        print(row)
-    in_file.close()
-
-
-# using_csv()
+    with open('languages.csv', 'r', newline='') as in_file:
+        in_file.readline()  # Skip header
+        reader = csv.reader(in_file)
+        for row in reader:
+            print(row)
 
 
 def using_namedtuple():
     """Language file reader version using a named tuple."""
-    in_file = open('languages.csv', 'r', newline='')
-    file_field_names = in_file.readline().strip().split(',')
-    print(file_field_names)
-    # Language will be a new subclass of the tuple data type class
-    Language = namedtuple('Language', 'name, typing, reflection, year')
-    reader = csv.reader(in_file)  # use default dialect, Excel
+    with open('languages.csv', 'r', newline='') as in_file:
```

Comment:
You could consider setting up pylint to get all the warnings in your code for anything that doesn't meet  PEP8 guidelines, such as not specifying an encoding when opening a file.